biosys/apps/main/tests/api/test_species_observation.py

```diff
@@ -572,337 +574,364 @@ def test_species_name(self):
+            'Longitude': 115.75,
+            'When': '2018-01-31'
+        }
 
-        self.readonly_user = User.objects.filter(username="readonly").first()
-        self.assertIsNotNone(self.custodian_2_user)
-        self.assertFalse(self.site_2.is_custodian(self.readonly_user))
-        self.assertFalse(self.site_1.is_custodian(self.readonly_user))
-        self.readonly_user.set_password(password)
-        self.readonly_user.save()
-        self.readonly_client = APIClient()
-        self.assertTrue(self.readonly_client.login(username=self.readonly_user.username, password=password))
+        # clear all records
+        ds.record_queryset.delete()
+        self.assertEquals(ds.record_queryset.count(), 0)
+        payload = {
+            "dataset": ds.pk,
+            "data": data
+        }
+        url = reverse('api:record-list')
+        self.assertEqual(
+            client.post(url, data=payload, format='json').status_code,
+            status.HTTP_201_CREATED
+        )
+        self.assertEquals(ds.record_queryset.count(), 1)
+        self.assertEquals(ds.record_queryset.first().species_name, 'Chubby Bat')
+
+    def test_update(self):
+        """
+        Test that name extraction after a PUT method
+        """
+        project = self.project_1
+        client = self.custodian_1_client
+        schema = self.schema_with_species_name()
+        ds = self._create_dataset_with_schema(project, client, schema, dataset_type=Dataset.TYPE_SPECIES_OBSERVATION)
+        data = {
+            'Species Name': 'Chubby Bat',
+            'Latitude': -32.0,
+            'Longitude': 115.75,
+            'When': '2018-01-31'
+        }
+        payload = {
+            "dataset": ds.pk,
+            "data": data
+        }
+        url = reverse('api:record-list')
+        self.assertEqual(
+            client.post(url, data=payload, format='json').status_code,
+            status.HTTP_201_CREATED
+        )
+        self.assertEquals(ds.record_queryset.count(), 1)
+        record = ds.record_queryset.first()
+        self.assertEquals(record.species_name, 'Chubby Bat')
 
-        self.anonymous_client = APIClient()
+        # update the species_name
+        data = {
+            'Species Name': ' Canis lupus ',
+            'Latitude': -32.0,
+            'Longitude': 115.75,
+            'When': '2018-01-31'
+        }
+        payload = {
+            "dataset": ds.pk,
+            "data": data
+        }
+        url = reverse('api:record-detail', kwargs={"pk": record.pk})
+        self.assertEqual(
+            client.put(url, data=payload, format='json').status_code,
+            status.HTTP_200_OK
+        )
+        record.refresh_from_db()
+        self.assertEqual(record.species_name, 'Canis lupus')
 
-    def test_create(self):
+    def test_patch(self):
         """
-        Test that the species name is extracted from the data and saved in DB
-        :return:
+        Test that name extraction after a PUT method
```

Comment:
PUT or PATCH? also you're using a post method below.

Reply:
The post is just to create the initial record